File: interview/retest/leetcode236/dfs_236.cpp
// /**
//  * Definition for a binary tree node.
//  * struct TreeNode {
//  *     int val;
//  *     TreeNode *left;
//  *     TreeNode *right;
//  *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
//  * };
//  */
// class Solution {
// public:
//     TreeNode* lowestCommonAncestor(TreeNode* root, TreeNode* p, TreeNode* q) {
//         if(!root||root==p||root==q)return root;
//         TreeNode*lf=lowestCommonAncestor(root->left,p,q),*rf=lowestCommonAncestor(root->right,p,q); 

//         if(lf&&rf)return root;
//         else if(lf)return lf;
//         else return rf;
//     }
// };

File: interview/retest/leetcode236/dfs_236.go
// /**
//  * Definition for a binary tree node.
//  * type TreeNode struct {
//  *     Val int
//  *     Left *TreeNode
//  *     Right *TreeNode
//  * }
//  */
//  func lowestCommonAncestor(root, p, q *TreeNode) *TreeNode {
//     if root==nil||root==p||root==q{
//         return root
//     }

//     lf:=lowestCommonAncestor(root.Left,p,q)
//     rf:=lowestCommonAncestor(root.Right,p,q)

//     if lf==nil{
//         return rf
//     }
//     if rf==nil{
//         return lf
//     }
//     return root

// }